doc/specs/Proto extensions-spec.md

```diff
@@ -0,0 +1,378 @@
+the users of the system - in this case, the installer should add their json files to the global folder:
+
+```
+C:\ProgramData\Microsoft\Windows Terminal\Fragments
+```
+
+Note: `C:\ProgramData` is a [known folder](https://docs.microsoft.com/en-us/previous-versions/windows/desktop/legacy/bb776911(v=vs.85))
+that apps should be able to access. It will be on us to create the folder `Windows Terminal` in `C:\ProgramData\Microsoft\Windows`
+for this to happen. 
+
+In the second case, the installation is only for the current user. For this case, the installer should add the
+json files to the local folder:
+
+```
+C:\Users\<user>\AppData\Local\Microsoft\Windows Terminal\Fragments
+```
+
+We will look through both folders mentioned above during profile generation. 
+
+#### The source field
+
+Currently, we allow users an easy way to disable/enable profiles that are generated from certain sources. For
+example, a user can easily hide all dynamic profiles with the source `"Windows.Terminal.Wsl"` if they wish to.
+To retain this functionality, we will add source fields to profiles we create through proto-extensions. 
+
+For full profiles that came from the *global* folder `C:\ProgramData\Microsoft\Windows\Terminal`,
+we will give the value `global` to the source field.
+
+For full profiles that came from the *local* folder `C:\Users\<user>\AppData\Local\Microsoft\Windows\Terminal`,
+we will give the value `local` to the source field.
+
+For full profiles that came from app extensions, we will give the value `app` to the source field.
+
+
+
+## UI/UX Design
+
+This feature will allow other installations a level of control over how their profiles look in Terminal. For example,
+if Ubuntu gets a new icon or a new font they can have those changes be reflected in Terminal users' Ubuntu profiles.
+
+## Capabilities
+
+### Accessibility
+
+This change should not affect accessibility.
+
+### Security
+
+Opening a profile causes its commandline argument to be automatically run. Thus, if malicious modifications are made
+to existing profiles or new profiles with malicious commandline arguments are added, users could be tricked into running
+things they do not want to.
+
+### Reliability
+
+This should not affect reliability - most of what its doing is simply layering json which we already do. 
+
+### Compatibility
+
+Again, there should not be any issues here - the user settings can still be layered after this layering for the user
+to have the final say. 
+
+### Performance, Power, and Efficiency
+
+Looking through the additional json files could negatively impact startup time. 
+
+## Potential Issues
+
+* An installer dumps a _lot_ of json files into the folder which we need to look through.
+* When a `.json` files is deleted, any new profiles that were generated from it remain in the user's settings file (though they no longer appear in the tab dropdown).

```

Comment:
That already happens - say I get a new profile from a json file, then delete the file, that new profile will not appear in the tab dropdown anymore (the stub sticks around in the user's settings file though, but that is also how the dynamic profiles work)

Reply:
Oh but I believe with dynamic profiles, we see that the user had a stub for a dynamic profile, but also that we _didn't_ generate the profile which that stub corresponds to. I _believe_ in that case, we actually insert `"hidden": true` into the user's settings for that profile, to auto-hide it. We should probably do that for the stubs we're generating here as well.

Reply:
Chatted with @DHowett , apparently we _don't_ change the `hidden` field in the user's settings in that case

Reply:
Yeah, we've only ever patched _one_ individual settings key, and that's `$schema`. For all the rest, we only insert profile blobs and then _ignore them_ when they don't exist.

If we set hidden:true, it would be hidden even after the profile came back. Consider how the WSL generator apparently fails to find WSL every other morning for some folks!

Reply:
Oh, my b. Must've been remembering an earlier version of that spec then. As long as the profile's gone, I'm cool with this 👍